homeassistant/components/plugwise/config_flow.py

```diff
@@ -109,13 +109,48 @@ async def async_step_user(self, user_input=None):
 
             if not errors:
                 await self.async_set_unique_id(api.gateway_id)
+                self._abort_if_unique_id_configured()
 
                 return self.async_create_entry(title=api.smile_name, data=user_input)
 
         return self.async_show_form(
-            step_id="user", data_schema=_base_schema(self.discovery_info), errors=errors
+            step_id="user",
+            data_schema=_base_schema(self.discovery_info),
+            errors=errors or {},
         )
 
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return PlugwiseOptionsFlowHandler(config_entry)
+
+
+class PlugwiseOptionsFlowHandler(config_entries.OptionsFlow):
+    """Plugwise option flow."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the Plugwise options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        api = self.hass.data[DOMAIN][self.config_entry.entry_id]["api"]
+        interval = DEFAULT_SCAN_INTERVAL["thermostat"]
+        if api.smile_type == "power":
+            interval = DEFAULT_SCAN_INTERVAL["power"]
```

Comment:
Can `smile_type` only be `power` or `thermostat`?

```suggestion
        interval = DEFAULT_SCAN_INTERVAL[api.smile_type]
```

Reply:
For now we aren't aware of anything Plugwise is cooking up differently. We have Thermostat originally for Anna and Adam as climate control systems and the P1's for DSMR (and upcoming very soon Stretch which is sort of a zigbee gateway for older switch plugs)

Reply:
Looks like it can be 
https://github.com/plugwise/Plugwise-Smile/blob/master/Plugwise_Smile/Smile.py#L123

Reply:
Good call @bdraco - it's different for the newly supported ones, adjusting accordingly

Reply:
> Looks like it can be
> https://github.com/plugwise/Plugwise-Smile/blob/master/Plugwise_Smile/Smile.py#L123

True that